src/lib/filter/adjudicators.ts

```diff
@@ -93,44 +120,55 @@ export const definesNamespaces = pipe(definedNamespaces, equals([]), not);
 
-export const definedAnnotations = pipe(binding => binding?.filters?.annotations, defaultTo({}));
+export const definedAnnotations = pipe((binding: Partial<Binding>) => binding?.filters?.annotations, defaultTo({}));
 export const definesAnnotations = pipe(definedAnnotations, equals({}), not);
 
-export const definedLabels = pipe(binding => binding?.filters?.labels, defaultTo({}));
+export const definedLabels = pipe((binding: Partial<Binding>) => binding?.filters?.labels, defaultTo({}));
 export const definesLabels = pipe(definedLabels, equals({}), not);
 
-export const definedEvent = pipe(binding => binding?.event, defaultTo(""));
-export const definesDelete = pipe(definedEvent, equals(Operation.DELETE));
+export const definedEvent = pipe((binding: Partial<Binding>): Event => {
+  if (binding.event) {
+    return binding.event;
+  } else {
+    throw TypeError("binding.event was undefined");
+  }
+}, defaultTo(""));
+export const definesDelete = pipe(definedEvent, equals(Event.DELETE));
 
-export const definedGroup = pipe(binding => binding?.kind?.group, defaultTo(""));
+export const definedGroup = pipe((binding): string => binding?.kind?.group, defaultTo(""));
 export const definesGroup = pipe(definedGroup, equals(""), not);
 
-export const definedVersion = pipe(binding => binding?.kind?.version, defaultTo(""));
+export const definedVersion = pipe(
+  (binding: Partial<Binding>): string | undefined => binding?.kind?.version,
+  defaultTo(""),
+);
 export const definesVersion = pipe(definedVersion, equals(""), not);
 
-export const definedKind = pipe(binding => binding?.kind?.kind, defaultTo(""));
+export const definedKind = pipe((binding): string => binding?.kind?.kind, defaultTo(""));
 export const definesKind = pipe(definedKind, equals(""), not);
 
-export const definedCategory = pipe(binding => {
-  // prettier-ignore
-  return (
-    binding.isFinalize ? "Finalize" :
-    binding.isWatch ? "Watch" :
-    binding.isMutate ? "Mutate" :
-    binding.isValidate ? "Validate" :
-    ""
-  );
+export const definedCategory = pipe((binding: Partial<Binding>) => {
+  return binding.isFinalize
+    ? "Finalize"
+    : binding.isWatch
+      ? "Watch"
+      : binding.isMutate
+        ? "Mutate"
+        : binding.isValidate
+          ? "Validate"
+          : "";
 });
 
-export const definedCallback = pipe(binding => {
-  // prettier-ignore
-  return (
-    binding.isFinalize ? binding.finalizeCallback :
-    binding.isWatch ? binding.watchCallback :
-    binding.isMutate ? binding.mutateCallback :
-    binding.isValidate ? binding.validateCallback:
-    null
-  );
+export const definedCallback = pipe((binding: Partial<Binding>) => {
+  return binding.isFinalize
+    ? binding.finalizeCallback
+    : binding.isWatch
+      ? binding.watchCallback
+      : binding.isMutate
+        ? binding.mutateCallback
+        : binding.isValidate
+          ? binding.validateCallback
+          : null;
```

Comment:
We can reintroduce `// prettier-ignore`, but this seems like a chance to discuss the value of an `EVENT_TYPE` style enum. Can a binding be multiple things at once?